main/i2c.c

```diff
@@ -0,0 +1,93 @@
+#include "i2c.h"
+#include "init.h"
+
+#include <stdio.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <pico/stdlib.h>
+#include <hardware/i2c.h>
+
+void init_i2c() {
+	i2c_init(I2C_PORT, 100 * 1000); // currently at 100kHz
+
+	// Initialize I2C pins - sda(16), scl(17)
+	gpio_set_function(SDA_PIN, GPIO_FUNC_I2C);
+	gpio_set_function(SCL_PIN, GPIO_FUNC_I2C);
+
+	gpio_pull_up(SDA_PIN);
+	gpio_pull_up(SCL_PIN);
+}
+
+int read_i2c(uint8_t addr, uint8_t *output, size_t len) {
+	// false - finished with bus
+	return i2c_read_blocking (I2C_PORT, addr, output, len, false);
```

Comment:
Ik denk dat er een onderscheid is tussen "blocking" en "blocking (maar niet voor freertos)", en ik weet niet of deze samen zal spelen met freertos?